src/main/java/core/basesyntax/service/operations/ReturnOperation.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class ReturnOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (Storage.fruits.containsKey(transaction.getFruit())
+                && transaction.getOperation() == FruitTransaction.Operation.RETURN) {
+            int newValue = Storage.fruits.get(transaction.getFruit())

```

Comment:
The calculation of the new value does not handle potential integer overflows. The sum of the current value and the transaction quantity should be checked for overflow before being stored.

Reply:
ok, done